contracts/ModelRepository.sol

```diff
@@ -73,14 +73,22 @@ contract ModelRepository {
     models.push(newModel);
   }
 
+  function incentiveCalculate(uint bounty, uint totalError, uint solvedError) returns(uint total) {
+    total = ((bounty/totalError) * solvedError);
+    return total;
+  }
+
   function evalGradient(uint gradientId, uint newModelError, bytes32[] newWeightsAddress) onlyByModelOwner(gradientId) onlyIfGradientNotYetEvaluated(gradientId) {
     grads[gradientId].newWeights.first = newWeightsAddress[0];
     grads[gradientId].newWeights.second = newWeightsAddress[1];
     grads[gradientId].newModelError = newModelError;
 
     Model model = models[grads[gradientId].modelId];
-    if (newModelError < model.bestError) {
-      uint incentive = ((model.bestError - newModelError) * model.bounty) / model.bestError;
+    if (newModelError < model.bestError && model.targetError < model.bestError) {
+      uint totalError = model.initialError - model.targetError;
+      uint solvedError = model.bestError - newModelError;
+
+      uint incentive = incentiveCalculate(model.bounty, totalError, solvedError);
```

Comment:
`calculateIncentive` might seem like a better fit if we compare it with the other function names

Reply:
Yes. Correct.